### PR TITLE
Add code sample to show how Hazelcast works with Pandas DataFrames.

### PR DESCRIPTION
**Pull request by meltemtokgoz**
Hi 👋

Fixes: https://github.com/hazelcast/hazelcast-python-client/issues/601

This PR adds a code sample to document how Pandas DataFrame can be used with Hazelcast Python Client. There are multiple ways to store the DataFrame in Hazelcast, I tried to list them all.

 I am new to open source, so I would really appreciate all comments if you have any suggesstion. 

Thanks in advance.

### CODE REVIEW

**Review by yuce — CHANGES_REQUESTED** [review]
Thanks for your contribution, here are a few ideas...

File: examples/packages/pandas_example.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import np as np
+import pandas as pd
+from matplotlib import pyplot as plt
+import hazelcast
+
+"""Create Hazelcast client connection"""
```
[the remark this answers is not in the document]
Comment (meltemtokgoz) [reply examples/packages/pandas_example.py]: I have converted single line comments to this format, see https://github.com/hazelcast/hazelcast-python-client/pull/604/commits/5f056213f517f4e5d0fae549b05f5ba7d1f6e442.

File: examples/packages/pandas_example.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    Convert to CSV: df = df.to_csv()
+    Convert to JSON: df = df.to_json()
+    Convert to dict: df = df.to_dict()
+
+    If you use these representation, you need to re-create DataFrame object as follows since they are stored as 
+    string in Hazelcast cluster. For example, if you are converting DataFrame to CSV, use following while reading: 
+
+    df = pd.from_csv(StringIO(patients.get("patient-1")))
+    """
+    patients.put(f"{pid}", df.to_json())
```
[the remark this answers is not in the document]
Comment (meltemtokgoz) [reply examples/packages/pandas_example.py]: Now string value is wrapped in HazelcastJsonValue, see https://github.com/hazelcast/hazelcast-python-client/pull/604/commits/5f056213f517f4e5d0fae549b05f5ba7d1f6e442.

File: examples/packages/pandas_example.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            "blood_pressure": np.random.randint(80, 120, size=(75,)),
+            "heart_rate": np.random.randint(60, 100, size=(75,)),
+        },
+        index=pd.date_range("1/1/2023", periods=75, freq="H"),
+    )
+    """
+    There are multiple serialization methods for DataFrame. Hazelcast uses pickle serialization as default
+    for various Python objects. So, you can simply put your data directly using default pickle serialization.
+
+    patients.put("1", df)
```
[the remark this answers is not in the document]
Comment (meltemtokgoz) [reply examples/packages/pandas_example.py]: Yes right, it makes more sense to use integer, see https://github.com/hazelcast/hazelcast-python-client/pull/604/commits/5f056213f517f4e5d0fae549b05f5ba7d1f6e442.

File: examples/packages/pandas_example.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    Notice that these methods returns string representation of DataFrame in requested format.
+
+    Convert to CSV: df = df.to_csv()
+    Convert to JSON: df = df.to_json()
+    Convert to dict: df = df.to_dict()
+
+    If you use these representation, you need to re-create DataFrame object as follows since they are stored as 
+    string in Hazelcast cluster. For example, if you are converting DataFrame to CSV, use following while reading: 
+
+    df = pd.from_csv(StringIO(patients.get("patient-1")))
```
[the remark this answers is not in the document]
Comment (meltemtokgoz) [reply examples/packages/pandas_example.py]: Fixed at https://github.com/hazelcast/hazelcast-python-client/pull/604/commits/5f056213f517f4e5d0fae549b05f5ba7d1f6e442.

File: examples/packages/pandas_example.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
+import np as np
+import pandas as pd
+from matplotlib import pyplot as plt
+import hazelcast
+
+"""Create Hazelcast client connection"""
+client = hazelcast.HazelcastClient()
+
+"""Get a IMap for storing patient's DataFrame"""
```
[the remark this answers is not in the document]
Comment (meltemtokgoz) [reply examples/packages/pandas_example.py]: Fixed at https://github.com/hazelcast/hazelcast-python-client/pull/604/commits/5f056213f517f4e5d0fae549b05f5ba7d1f6e442.

File: examples/packages/pandas_example.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+for pid in range(0, 50):
+    """
+    Create a random DataFrame.
+    """
+    df = pd.DataFrame(
+        data={
+            "blood_pressure": np.random.randint(80, 120, size=(75,)),
+            "heart_rate": np.random.randint(60, 100, size=(75,)),
+        },
+        index=pd.date_range("1/1/2023", periods=75, freq="H"),
```
[the remark this answers is not in the document]
Comment (meltemtokgoz) [reply examples/packages/pandas_example.py]: Converted to this format at https://github.com/hazelcast/hazelcast-python-client/pull/604/commits/5f056213f517f4e5d0fae549b05f5ba7d1f6e442.

File: examples/packages/pandas_example.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    )
+    """
+    There are multiple serialization methods for DataFrame. Hazelcast uses pickle serialization as default
+    for various Python objects. So, you can simply put your data directly using default pickle serialization.
```
[the remark this answers is not in the document]
Comment (meltemtokgoz) [reply examples/packages/pandas_example.py]: Yes, the description is general information about code sample and can be placed at the top, see https://github.com/hazelcast/hazelcast-python-client/pull/604/commits/5f056213f517f4e5d0fae549b05f5ba7d1f6e442.

**Comment by meltemtokgoz** [comment]
Thank for the review @yuce. I tried to fix the comments, let me know if I miss anything. 👍🏻

File: examples/packages/pandas_example.py [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+Create from dict object: df = pd.DataFrame(patients.get(1))
+
+In addition to methods above, you can write your own custom serializer for DatFrame objects. For more information about
+Pyton client serialization methods, see https://hazelcast.readthedocs.io/en/stable/serialization.html#
+"""
+
+import hazelcast
+from hazelcast.core import HazelcastJsonValue
+from matplotlib import pyplot as plt
+import np as np
```
Comment (yuce) [inline examples/packages/pandas_example.py:30]: This should be `import numpy as np`.

**Review by yuce — CHANGES_REQUESTED** [review]
Thanks, looks good except an import error.

Could you also move this file to `examples/pandas/example.py`?

File: examples/packages/pandas_example.py (hunk 8 above)
Comment (meltemtokgoz) [reply examples/packages/pandas_example.py]: Thanks, fixed now https://github.com/hazelcast/hazelcast-python-client/pull/604/commits/e1d903640dc9d5c3146b084577695a1b0d5bcc1f.

**Review by yuce — APPROVED** [review]

**Comment by yuce** [comment]
@meltemtokgoz Thanks for your contribution!

**Pull request merged**